components/threading/global_threads/thread_item/thread_item.tsx

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo, ComponentProps, useCallback, MouseEvent} from 'react';
+import {FormattedMessage} from 'react-intl';
+import classNames from 'classnames';
+
+import './thread_item.scss';
+
+import Badge from 'components/widgets/badges/badge';
+import Timestamp from 'components/timestamp';
+import Avatars from 'components/widgets/users/avatars';
+
+import ThreadMenu from '../thread_menu';
+
+import {THREADING_TIME} from '../../common/options';
+
+type Props = {
+    participants: ComponentProps<typeof Avatars>['users'],
+    totalParticipants?: number;
+    name: string,
+    channelName: string,
+    previewText: string,
+
+    lastReplyAt: ComponentProps<typeof Timestamp>['value'],
+    newReplies: number,
+    newMentions: number,
+    totalReplies: number,
+
+    isFollowing: boolean,
+    isSaved: boolean,
+    isSelected: boolean,
+
+    actions: {
+        select: () => void,
+        openInChannel: () => void,
+    },
+} & Pick<ComponentProps<typeof ThreadMenu>, 'actions'>;
+
+const ThreadItem = ({
+    participants,
+    totalParticipants,
+    name,
+    channelName,
+    previewText,
+
+    lastReplyAt,
+    newReplies,
+    newMentions,
+    totalReplies,
+
+    isSelected,
+    isSaved,
+    isFollowing,
+
+    actions,
+
+}: Props) => {
+    return (
+        <div
+            className={classNames('ThreadItem', {
+                'has-unreads': newReplies,
+                'is-selected': isSelected,
+            })}
+            tabIndex={0}
+            onClick={actions.select}
+        >
```

Comment:
Here's another case I'd considering using an `<article>` tag. That would also allow you to 'reset' the heading level to `<h1>` and not have to worry about counting heading levels to try to get the correct semantic level here.